research/order_level_classifier/model_evaluation/binary_model_evaluation_ami_traps.py

```diff
@@ -0,0 +1,180 @@
+"""
+Author: Aditya Jain
+Date started: April 18, 2024
+About: Evaluation of AMI-GBIF trained binary classifier on AMI-Traps data
+"""
+
+import glob
+import json
+import os
+import pathlib
+from pathlib import Path
+
+import torch
+
+# 3rd party packages
+from dotenv import load_dotenv
+from PIL import Image
+
+from src.classification.model_inference import ModelInference
+
+# Load secrets and config from optional .env file
+load_dotenv()
+
+
+def _get_insect_crops_and_labels(insect_crops_dir: pathlib.PosixPath):
+    """Get all insect crops and label information"""
+
+    insect_crops = glob.glob(str(insect_crops_dir / "*.png"))
+    with open(insect_crops_dir / "binary_labels.json") as f:
+        insect_labels = json.load(f)
+
+    return insect_crops, insect_labels
+
+
+def _update_evaluation_metrics(
+    gt_label: str,
+    pred: str,
+    tp: int,
+    tn: int,
+    fp: int,
+    fn: int,
+    gt_moths: int,
+    gt_nonmoths: int,
+):
+    """Update evaluation metrics"""
+
+    if gt_label == "Moth":
+        if pred == "Moth":
+            tp += 1
+        else:
+            fn += 1
+        gt_moths += 1
+    elif gt_label == "Non-Moth":
+        if pred == "Non-Moth":
+            tn += 1
+        else:
+            fp += 1
+        gt_nonmoths += 1
+    else:
+        raise Exception("Unknown binary label for an insect crop!")
+
+    return tp, tn, fp, fn, gt_moths, gt_nonmoths
+
+
+def _get_aggregated_metrics(
+    tp: int, tn: int, fp: int, fn: int, gt_moths: int, gt_nonmoths: int
+):
+    """Get aggregated metrics"""
+
+    total_crops = gt_moths + gt_nonmoths
+    accuracy = round((tp + tn) / (tp + tn + fp + fn) * 100, 2)
+    precision = round((tp) / (tp + fp) * 100, 2)
+    recall = round((tp) / (tp + fn) * 100, 2)
+    fscore = round((2 * precision * recall) / (precision + recall), 2)
+
+    return total_crops, accuracy, precision, recall, fscore
+
+
+def binary_model_evaluation(
+    run_name: str,
+    model: str,
+    model_type: str,
+    model_dir: str,
+    category_map_key_to_id: str,
+    category_map_key_to_name: str,
+    insect_crops_dir: str,
+    skip_small_crops: bool = False,
+    min_crop_dim: int = 0,
+):
+    """Main function for binary model evaluation"""
+    print(f"Evaluating model for the run {run_name}.")
+
+    # Get the environment variables
+    device = "cuda" if torch.cuda.is_available() else "cpu"
+    print(f"Device {device} is available.")
+    model_checkpoint_path = str(Path(model_dir) / model)
+    category_map_key_to_id_path = str(Path(model_dir) / category_map_key_to_id)
+    category_map_key_to_name_path = str(Path(model_dir) / category_map_key_to_name)
```

Comment:
If you are indeed using these as paths, I'm pretty sure you shouldn't need to convert them to string, you just have to adjust `ModelInference()`to be able to use `Union[str, Path]` as a type, instead of just `str`